File: api/src/main/java/com/example/movie/presentation/controller/MovieController.java
package com.example.movie.presentation.controller;

import com.example.movie.application.dto.MoviesNowShowingDetail;
import com.example.movie.application.service.MovieService;
import com.example.response.BaseResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.List;

@RestController
@RequiredArgsConstructor
public class MovieController {

    private final MovieService movieService;

    @GetMapping("/v1/movies/now-showing")
    public BaseResponse<List<MoviesNowShowingDetail>> getMoviesNowShowing() {
        List<MoviesNowShowingDetail> response = movieService.getNowShowingMovies(LocalDateTime.now());
        return new BaseResponse<>(response);
    }

}

File: src/main/kotlin/com/example/demo/presentation/controller/MovieController.kt
package com.example.demo.presentation.controller

import com.example.demo.application.dto.MovieDto
import com.example.demo.application.service.MovieService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class MovieController(private val movieService: MovieService) {

    @GetMapping("/movies/now-showing")
    fun getNowShowingMovies(
        @RequestParam title: String?,
        @RequestParam genre: String?
    ): List<MovieDto> {
        return movieService.getNowShowingMovies(title, genre)
    }

    @GetMapping("/movies/now-showing/cached")
    fun getNowShowingMoviesWithCaching(
        @RequestParam title: String?,
        @RequestParam genre: String?
    ): List<MovieDto> {
        return movieService.getNowShowingMoviesWithCaching(title, genre)
    }
} 